src/settings_gui.cpp

```diff
@@ -216,13 +216,19 @@ struct GameOptionsWindow : Window {
 		switch (widget) {
 			case WID_GO_CURRENCY_DROPDOWN: { // Setup currencies dropdown
 				*selected_index = this->opt->locale.currency;
-				StringID *items = BuildCurrencyDropdown();
 				uint64 disabled = _game_mode == GM_MENU ? 0LL : ~GetMaskOfAllowedCurrencies();
 
 				/* Add non-custom currencies; sorted naturally */
-				for (uint i = 0; i < CURRENCY_END; items++, i++) {
+				for (const CurrencySpec &cur : _currency_specs) {
+					int i = &cur - _currency_specs;
 					if (i == CURRENCY_CUSTOM) continue;
-					list.emplace_back(new DropDownListStringItem(*items, i, HasBit(disabled, i)));
+					if (cur.code.empty()) {
+						list.emplace_back(new DropDownListStringItem(cur.name, i, HasBit(disabled, i)));
+					} else {
+						SetDParam(0, cur.name);
+						SetDParamStr(1, cur.code.c_str());
```

Comment:
The `.c_str()` is not needed with `SetDParamStr`. Like in line 319.

Reply:
Good point, it was needed with DropDownListParamStringItem :-)